hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/record/AbstractNearCacheRecord.java

```diff
@@ -42,6 +42,7 @@
             AtomicLongFieldUpdater.newUpdater(AbstractNearCacheRecord.class, "recordState");
 
     protected long creationTime = TIME_NOT_SET;
+    protected int partitionId;
```

Comment:
So you think we should make all three fields `volatile`? They are just set once, but unfortunately lazily after the instance creation.